src/js_errors.rs

```diff
@@ -46,6 +50,19 @@ pub struct JSError {
   pub frames: Vec<StackFrame>,
 }
 
+impl From<DenoError> for JSError {
+  fn from(error: DenoError) -> Self {
+    JSError {
+      message: [error.kind_str(), ": ", error.description()].join(""),
+      frames: vec![StackFrame {
+        script_name: "<unknown>".to_string(),
+        function_name: "".to_string(),
+        ..StackFrame::default()
+      }],
+    }
+  }
+}
```

Comment:
Oops - I had written this comment days ago - but I guess I forgot to actually send it.

Reply:
Slightly confused by the comments here... the error happens in `Isolate::execute_mod`, which is caused by an `.unwrap()` and happens on the main thread. I added this `From` impl since the Result error type for `execute_mod` is `JSError`...

That being said, I just checked the code again. It seems that in `resolve_cb` there is another `.unwrap`. However that crash is never triggered since the TS compiler would throw an error first when it is building its own dependency graph (which also causes #1496).

Reply:
I think I understand what it means now... Well it is a different scenario: the TS compiler swallows the error only when it is resolving `import` statements in each files to compile, but not in the entry file. `deno somefile.ts` this `somefile.ts` is the entry file, and its resolution is before even running the `compile_sync` and thus without compiler involvement